src/os/Transaction.h

```diff
@@ -291,6 +291,12 @@ class Transaction {
   Transaction(const Transaction& other) = default;
   Transaction& operator=(const Transaction& other) = default;
 
+  ~Transaction() {
+    ceph_assert(on_applied.empty());
+    ceph_assert(on_commit.empty());
+    ceph_assert(on_applied_sync.empty());
+  }
```

Comment:
Was this change reverted intentionally in 2add894a6ba6cc62535a82ca5cb20d8bac2c9ff7?

Reply:
Yes but I don't have any strong feelings towards this commit. It can be useful in testing but the `unique_ptr` makes it not-so-important.

Reply:
If so, 9eabc62bff1e8d679f7d14b6378861d388cc9090 can be dropped.